### PR TITLE
Loading Image with Glide in Android

### LINKED ISSUE #2771: Loading Image with Glide in Android
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article higher ranking.*

### Introduction paragraph (2-3 paragraphs):
Please write the Introductory paragraph(s), that would be included in your article. We will use this writing snippet to help us assess overall quality before approval. We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.
Take your time and write the content as you would intend to get it published. 

### Key takeaways: 
What are the 3-5 most important things the reader should understand or be able to do after reading this article? Use this area to get your ideas down on the bulk of your article or tutorial.

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Collince-Okeyo**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
This fixes #2771

### CODE REVIEW

**Comment by Ericgacoki** [comment]
Greetings @Collince-Okeyo,
Welcome to Section's Eng-Ed program. We're excited to have you on board. Thank you for submitting your `first article` for review.
I'll be reviewing it to check and help you fix any errors before being published!

**Comment by Ericgacoki** [comment]
I have a few suggestions to make before we proceed:

1. Please always use the template provided when opening a new PR.
2. Seems like you forgot to add author details 🤭. Please create your author folder in the `/content/authors` directory. This folder holds your author bio and an avatar. Feel free to check how other students have [structured theirs](https://github.com/section-engineering-education/engineering-education/tree/master/content/authors).
3. Your article folder should contain a `600 X 400px` image named `hero.png` (other image formats like jpeg and jpg are also allowed). This is used as the preview image once your article is published. Again, you can check out [these docs](https://github.com/section-engineering-education/engineering-education/tree/master/new_contributors) for more info about first-time contribution.
4. I'd recommend you to always open a PR from another branch other than the master branch. This isn't a big problem though. Just be super careful to only edit files related to this article.

Am going to add the template mentioned in (1) above and link this PR to your suggestion form (issue). Please check the necessary check-boxes. Thank you!

**Comment by Collince-Okeyo** [comment]
> Greetings @Collince-Okeyo,
> Welcome to Section's Eng-Ed program. We're excited to have you on board. Thank you for submitting your `first article` for review.
> I'll be reviewing it to check and help you fix any errors before being published!

Thank you @Ericgacoki am glad to have you as my Peer reviewer and this being my first article am much optimistic that you will guide me through

**Comment by Ericgacoki** [comment]
@Collince-Okeyo, ping me when you're done. Feel free to seek help (here or DM on Slack) 🙂.

**Comment by Ericgacoki** [comment]
Please rename your author image to `avatar.jpeg`.
And we currently don't support `.mp4` file formats yet. Kindly replace it with a `GIF` image.

**Comment by Collince-Okeyo** [comment]
> @Collince-Okeyo, ping me when you're done. Feel free to seek help (here or DM on Slack) slightly_smiling_face.

Have added author folder and hero image

**Comment by Ericgacoki** [comment]
> Please rename your author image to `avatar.jpeg`.
> And we currently don't support `.mp4` file formats yet. Kindly replace it with a `GIF` image.

Remember to check the necessary boxes on the article checklist.
```
unchecked box:
- [ ]
checked box:
- [x]
```

**Comment by Collince-Okeyo** [comment]
> Please rename your author image to `avatar.jpeg`.
> And we currently don't support `.mp4` file formats yet. Kindly replace it with a `GIF` image.

I tried pushing the changes above and there was an error...
collince@ramg-hp-elitebook-2560p:~/engineering-education$ git push origin
To https://github.com/Collince-Okeyo/engineering-education.git
 ! [rejected]            master -> master (non-fast-forward)
error: failed to push some refs to 'https://github.com/Collince-Okeyo/engineering-education.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

**Comment by Ericgacoki** [comment]
As the error reads, your `local branch` is behind the `remote` one. Remember I made some changes in your bio file.

> WARNING:
If you've already made some changes, running the commands below will revert them. A clever way to escape this is to copy your updated files elsewhere on your machine and return them when you're done.

Switch to your working branch (if you haven't already) and run:
```
git pull origin
```
or
```
git pull upstream
```
Return the updated files then run:
```
git add .
```
```
git commit -m "commit message"
```
followed by:
```
git push
```
or
```
git push origin
```

**Comment by Ericgacoki** [comment]
I think the above commit doesn't reflect the commit message. I mean, I can't see the renamed files.

**Comment by Ericgacoki** [comment]
Good work Collince. Thanks for the updates.

**Comment by Ericgacoki** [comment]
> Good work Collince. Thanks for the updates.

Moving on, we'll mainly focus on the article file (index.md). Please keep other files as they are, unless requested to change/edit them. :rocket:

File: content/articles/loading-image-with-glide-in-android/index.md [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+
+To understand this tutorial, the reader should:
+
+- Have Android Studio installed
+
+- Have an understanding of Kotlin programming language and XML.
+
+- Have a basic understanding of image loading in android applications.
+
+- Be familiar with `ViewBinding`.
```
Comment (Ericgacoki) [inline content/articles/loading-image-with-glide-in-android/index.md:16]: Consider removing the extra blank lines. Do the same in the entire article where applicable.

File: content/articles/loading-image-with-glide-in-android/index.md [hunk 2]
```diff
@@ -0,0 +1,605 @@
...
+
+<androidx.constraintlayout.widget.ConstraintLayout
+
+xmlns:android="http://schemas.android.com/apk/res/android"
+
+xmlns:app="http://schemas.android.com/apk/res-auto"
+
+xmlns:tools="http://schemas.android.com/tools"
+
+android:layout_width="match_parent"
```
Comment (Ericgacoki) [inline content/articles/loading-image-with-glide-in-android/index.md:125]: You can improve code formatting by removing the blank lines in between the code scopes. Ensure that the correct indent is adhered to as well.

**Review by Ericgacoki — CHANGES_REQUESTED** [review]
Hello @Collince-Okeyo, check out the comments attached below concerning the format of your article.
Please fix them before we proceed. Ping me when you're done.

**Review by Ericgacoki — APPROVED** [review]

File: content/articles/loading-image-with-glide-in-android/index.md (hunk 2 above)
Comment (Collince-Okeyo) [reply content/articles/loading-image-with-glide-in-android/index.md]: Hello

**Comment by Ericgacoki** [comment]
Hello @Collince-Okeyo, I have made a few changes for flow and formatting. Be sure to check them out and remember to `pull` the changes ✌🏽.

File: content/articles/loading-image-with-glide-in-android/index.md [hunk 3]
```diff
@@ -0,0 +1,306 @@
+In this tutorial, we are going to learn how to use the Glide library to load images either from the internet (URL) or drawable image file into an `ImageView` in Android.
+
+### Prerequisites
+To follow along with this tutorial, the reader should:
+- Have Android Studio installed
+- Have an understanding of `Kotlin` programming language and `XML`
+- Have a basic understanding of image loading in android applications
+- Be familiar with `ViewBinding`
```
Comment (Ericgacoki) [inline content/articles/loading-image-with-glide-in-android/index.md:8]: Provide necessary links for the key tools and skills mentioned above.

File: content/articles/loading-image-with-glide-in-android/index.md [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+
+### Goal
+By the end of this tutorial, the reader will be able to:
+- Understand what Glide is
+- Add an `ImageView` to an `XML` layout
+- Load images using Glide
+
+### Introduction  
+Loading images in Android applications was hard and developers had to solve this problem by building various visual representation libraries such as Glide, Picasso, Coil image loader, and Fresco. In this article, we will discuss Glide in details.
```
Comment (Ericgacoki) [inline content/articles/loading-image-with-glide-in-android/index.md:17]: Provide links for these libraries as well.

File: content/articles/loading-image-with-glide-in-android/index.md [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+Glide.with(this)
+    .load(R.drawable.image)
+    .into(binding.imageView)
+}
+```
+### Transformations
+In Glide, transformations take a resource, mutate it, and return the mutated resource. They are also used to crop or apply filters to `Bitmaps`.
+
+### a) Center Crop
+Is implemented by using the `centerCrop()` method.
```
Comment (Ericgacoki) [inline content/articles/loading-image-with-glide-in-android/index.md:201]: Give a short description/definition before implementing a concept. Do the same for the other transformations.

File: content/articles/loading-image-with-glide-in-android/index.md [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+    override fun onResourceReady(
+    resource: GifDrawable,
+    transition: Transition<in GifDrawable>?
+) {
+    resource.start()
+    binding.imageView.setImageDrawable(resource)
+}
+})
+}
+```
```
Comment (Ericgacoki) [inline content/articles/loading-image-with-glide-in-android/index.md:298]: This block of code is not well-formatted. Rectify that.

**Review by Ericgacoki — CHANGES_REQUESTED** [review]

File: content/articles/loading-image-with-glide-in-android/index.md (hunk 6 above)
Comment (Collince-Okeyo) [reply content/articles/loading-image-with-glide-in-android/index.md]: Am done with the corrections

Comment (Ericgacoki) [reply content/articles/loading-image-with-glide-in-android/index.md]: Let me take a look 👁.👁

**Review by Ericgacoki — APPROVED** [review]

File: content/articles/loading-image-with-glide-in-android/index.md [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+    app:layout_constraintTop_toBottomOf="@+id/buttonUrl" />
+<Button
+    android:id="@+id/buttonScale2"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginStart="4dp"
+    android:text="CenterCrop"
+    app:layout_constraintBottom_toBottomOf="@+id/buttonScaling1"
+    app:layout_constraintStart_toEndOf="@+id/buttonScaling1"
+    app:layout_constraintTop_toTopOf="@+id/buttonScaling1" />
```
Comment (Ericgacoki) [inline content/articles/loading-image-with-glide-in-android/index.md:125]: @Collince-Okeyo, these two buttons are a little confusing. Their ids do not match the text or the action they imply.
Consider **refactoring** the ids, for instance, `buttonFitCenter` is more relevant for the `FitCenter` button.

File: content/articles/loading-image-with-glide-in-android/index.md [hunk 8]
```diff
@@ -0,0 +1,334 @@
...
+
+#### Loading an image from URL
+Glide allows the display of images from links (URL) without downloading them.
+
+Add the following lines of code in the `MainActivity.kt` file:
+
+```kotlin
+binding.buttonUrl.setOnClickListener {
+Glide.with(this)
+    .load(resizeImage)
```
Comment (Ericgacoki) [inline content/articles/loading-image-with-glide-in-android/index.md:189]: This seems like an URL variable. Did you forget to declare and initialize it?

File: content/articles/loading-image-with-glide-in-android/index.md [hunk 9]
```diff
@@ -0,0 +1,334 @@
...
+```
+
+#### Loading Drawable Image
+Glide allows loading of `Drawable` image files into `ImageViews`.
+In the `MainActivity.kt` file, add the following lines of code:
+
+```kotlin
+binding.buttonDrawable.setOnClickListener {
+Glide.with(this)
+    .load(R.drawable.image)
```
Comment (Ericgacoki) [inline content/articles/loading-image-with-glide-in-android/index.md:201]: I've seen quite a number of distinct images used in this project. Running this code will result in an `unresolved reference exception`. You can provide a link to your `drawable` directory on GitHub where readers can download these images.

**Review by Ericgacoki — CHANGES_REQUESTED** [review]

**Comment by Ericgacoki** [comment]
@Collince-Okeyo, 🤦‍♂️ seems like you `reverted` my changes! Did you `git pull` before committing?

**Comment by Ericgacoki** [comment]
I've just returned the latest clean copy.
This time be sure to `pull` and REDO the changes requested. Please let's avoid reverting/ignoring each other's changes. Otherwise, the review process might take **longer** than expected.
Good luck!

**Comment by Ericgacoki** [comment]
Awesome 🥳 🥳... Thanks for the updates!

**Review by Ericgacoki — APPROVED** [review]

**Comment by Ericgacoki** [comment]
@Collince-Okeyo, tech review - ✅ `done` .
Your article will be `ready for final review` after passing the plagiarism check!

**Comment by Collince-Okeyo** [comment]
> Awesome partying_face partying_face... Thanks for the updates!



> Awesome partying_face partying_face... Thanks for the updates!

Great bro your comments are really encouraging and you have made me learn a lot.

**Comment by Collince-Okeyo** [comment]
> @Collince-Okeyo, tech review - white_check_mark `done` .
> Your article will be `ready for final review` after passing the plagiarism check!

Thanks for updates

**Comment by hectorkambow** [comment]
@Collince-Okeyo 
I'll be running a final review on your PR when ready.
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was way higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[Loading Image with Glide in Android #3189.pdf](https://github.com/section-engineering-education/engineering-education/files/6963473/Loading.Image.with.Glide.in.Android.3189.pdf)

**Comment by Ericgacoki** [comment]
Hello @Collince-Okeyo, thanks for the updates. Just wanted to know if you have checked plagiarism after making the changes.

**Comment by Collince-Okeyo** [comment]
> Hello @Collince-Okeyo, thanks for the updates. Just wanted to know if you have checked plagiarism after making the changes.

Hello too @Ericgacoki , Have checked the plagiarism.

**Comment by Collince-Okeyo** [comment]
> @Collince-Okeyo
> I'll be running a final review on your PR when ready.
> Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was way higher than we typically accept.
> Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
> Be sure to see our resources page to see more info on plagiarism and what is considered as such.
> 
> [Loading Image with Glide in Android #3189.pdf](https://github.com/section-engineering-education/engineering-education/files/6963473/Loading.Image.with.Glide.in.Android.3189.pdf)

Have done the plagiarism check

File: content/articles/loading-image-with-glide-in-android/index.md [hunk 10]
```diff
@@ -0,0 +1,351 @@
...
+Before the finish of this tutorial, the reader should be able to:
+- Understand what [Glide](https://bumptech.github.io/glide/) is.
+- Add an [ImageView](https://developer.android.com/reference/android/widget/ImageView) to an `XML` layout
+- Load images using the `Glide` library.
+
+### Introduction  
+Loading images in Android applications was difficult and developers had to solve this problem by building various visual representation libraries such as `Glide`, [Picasso](https://square.github.io/picasso/), [Coil](https://coil-kt.github.io/coil/getting_started/) image loader, and [Fresco](https://frescolib.org/docs/). In this article, we will discuss more about the Glide library in detail.
+
+### What is Glide?
+As indicated by true documentation, Glide is a quick and productive image loading library for Android focused on smooth scrolling.
```
Comment (Ericgacoki) [inline content/articles/loading-image-with-glide-in-android/index.md:37]: What does true documentation mean?

File: content/articles/loading-image-with-glide-in-android/index.md [hunk 11]
```diff
@@ -0,0 +1,351 @@
...
+<!--more-->
+### Prerequisites
+To follow through this tutorial, the reader should:
+- Have [Android Studio](https://developer.android.com/studio) installed.
+- Have a good understanding of the [Kotlin](https://kotlinlang.org/) programming language and [XML](https://developer.android.com/guide/topics/ui/declaring-layout) layout.
+- Have a basic understanding of image loading in Android applications.
+- Be familiar with [ViewBinding](https://developer.android.com/topic/libraries/view-binding).
+
+### Goal
+Before the finish of this tutorial, the reader should be able to:
```
Comment (Ericgacoki) [inline content/articles/loading-image-with-glide-in-android/index.md:28]: Your last commit seems to have messed up with grammar a little bit.

File: content/articles/loading-image-with-glide-in-android/index.md [hunk 12]
```diff
@@ -0,0 +1,351 @@
...
+```kotlin
+binding.buttonDrawable.setOnClickListener {
+Glide.with(this)
+    .load(R.drawable.image)
+    .into(binding.imageView)
+}
+```
+
+### Transformations
+Transformations in Glide involve taking a resource, transforming it, and returning the changed resource. Transformations are also used to crop and apply filters on images.
```
Comment (Ericgacoki) [inline content/articles/loading-image-with-glide-in-android/index.md:222]: 😠 You can't explain a word with itself...check on this too.

**Review by Ericgacoki — APPROVED** [review]

**Comment by hectorkambow** [comment]
Plagiarism check did improve @Ericgacoki

**Comment by ahmadmardeni1** [comment]
Great work gentlemen!
Welcome to Section EngEd and congrats on your first article 🎉 @Collince-Okeyo 

Thank you for the review @Ericgacoki

**Pull request merged**